### PR TITLE
DM-45226: Modify nan/inf usage to support numpy 2.0

### PR DESCRIPTION
**Pull request by timj**

### CODE REVIEW

File: python/lsst/analysis/tools/actions/vector/selectors.py [hunk 1]
```diff
@@ -517,7 +517,7 @@ def __call__(self, data: KeyedData, **kwargs) -> Vector:
             case _:
                 bands = None
         if bands:
-            mask = np.in1d(data[self.vectorKey], bands)
+            mask = np.isin(data[self.vectorKey], bands)
```
Comment (timj) [inline python/lsst/analysis/tools/actions/vector/selectors.py:520]: @bechtol you wrote this line originally. Does this change look okay to you? I think the difference between the two is that the former always returned a 1-D array but now it returns something in the same shape as the original.

Comment (kherner) [reply python/lsst/analysis/tools/actions/vector/selectors.py:520]: I'm sure you already read it but to post for the record, in1d is officially deprecated in 2.0+ and isin is the replacement. Is there any case where the shape preservation of isin() would be a problem?

[numpy documentation on in1d](https://numpy.org/doc/stable/reference/generated/numpy.in1d.html)

Comment (kherner) [reply python/lsst/analysis/tools/actions/vector/selectors.py:520]: Personally I can't think of anything but I am far from 100% knowledgeable on such things.

**Review by kherner — APPROVED** [review]
See comments.

File: python/lsst/analysis/tools/actions/vector/selectors.py (hunk 1 above)
Comment (timj) [reply python/lsst/analysis/tools/actions/vector/selectors.py:520]: The issue is if `mask` is expected to be 1D but `data` is N-D, in which case the return from `isin` presumably has to be turned into 1-D. The mask created by `np.full` is going to be 1D.

Comment (timj) [reply python/lsst/analysis/tools/actions/vector/selectors.py:520]: Unfortunately this method is never called in tests.

Comment (timj) [reply python/lsst/analysis/tools/actions/vector/selectors.py:520]: Ok. I'm going to change this to call `.flatten()` to guarantee it's returning the same thing.

File: python/lsst/analysis/tools/actions/vector/selectors.py [hunk 2]
```diff
@@ -517,7 +517,7 @@ def __call__(self, data: KeyedData, **kwargs) -> Vector:
             case _:
                 bands = None
         if bands:
-            mask = np.in1d(data[self.vectorKey], bands)
+            mask = np.isin(data[self.vectorKey], bands).flatten()
```
Comment (kherner) [inline python/lsst/analysis/tools/actions/vector/selectors.py:520]: Fair enough.

**Pull request merged**